rocAL/source/readers/file_source_reader.cpp

```diff
@@ -137,12 +167,26 @@ int FileSourceReader::release() {
 }
 
 void FileSourceReader::reset() {
-    if (_shuffle) std::random_shuffle(_file_names.begin(), _file_names.end());
+    if (_shuffle)
+        std::random_shuffle(_all_shard_file_names_padded.begin(),
+                            _all_shard_file_names_padded.end());
+
+    if (_stick_to_shard == false)
+        increment_shard_id(); // Should work for both single and multiple shards
```

Comment:
Remove comment if its working

Reply:
comments added to make the understanding better